.cirrus.yml

```diff
@@ -59,6 +59,18 @@ validate_task:
         image_name: ${DEBIAN_CACHE_IMAGE_NAME}
     env:
         HOME: "/root"  # default unset, needed by golangci-lint.
+    go_modules_cache: &go_modules_cache
+        fingerprint_script: cat go.sum
+        folder: $GOPATH/pkg/mod
+    go_build_cache: &go_build_cache
+        fingerprint_script: cat go.sum
+        folder: $GOCACHE
+    golangci-lint_cache:
+        folder: /root/.cache/golangci-lint
+        reupload_on_changes: true
+        fingerprint_script:
+            - go version
+            - grep GOLANGCI_LINT_VERSION Makefile | head -1
     script: |
         git remote update
         make tools
```

Comment:
All of this could be cached, or maybe built into the VM image.

Reply:
Separating the caches between `validate_task` and `cross_task` helps at least in the sense that we don’t download `git-validation` and its dependencies (and hopefully the compilation is also avoided).

Not the best we can do, but getting closer to the point of diminishing returns.